### PR TITLE
Ros1 devel

### PR DESCRIPTION
**Pull request by pkowalsk1**

### CODE REVIEW

File: package.xml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+  <url type="website">https://husarion.com/</url>
+  <url type="repository">https://github.com/husarion/panther_description/tree/ros1</url>
+  <url type="bugtracker">https://github.com/husarion/panther_description/issues</url>
+
+  <buildtool_depend>catkin</buildtool_depend>
+
+  <exec_depend>joint_state_publisher</exec_depend>
+  <exec_depend>urdf</exec_depend>
+  <exec_depend>xacro</exec_depend>
```
Comment (Kotochleb) [inline package.xml:19]:
```suggestion
  <depend>gazebo_ros</depend>
  <depend>urdf</depend>
  <depend>xacro</depend>
```

File: package.xml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0"?>
+<package format="2">
```
Comment (Kotochleb) [inline package.xml:2]:
```suggestion
<package format="3">
```

File: package.xml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  <name>ros_components_description</name>
+  <version>0.1.0</version>
+  <description>URDF descriptions of components offered with Husarion robots</description>
+
+  <maintainer email="support@husarion.com">Husarion</maintainer>
+  <maintainer email="krzysztof.wojciechowski@husarion.com">Krzysztof Wojciechowski</maintainer>
```
Comment (Kotochleb) [inline package.xml:8]:
```suggestion
  <maintainer email="krzysztof.wojciechowski@husarion.com">Krzysztof Wojciechowski</maintainer>
  <maintainer email="pawel.kowalski@husarion.com">Pawel Kowalski</maintainer>
```

File: urdf/orbbec_astra.gazebo.xacro [hunk 4]
```diff
@@ -0,0 +1,88 @@
+<?xml version="1.0"?>
+<robot xmlns:xacro="http://www.ros.org/wiki/xacro">
+  <xacro:macro name="orbbec_astra" params="xyz rpy prefix:=front">
```
Comment (Kotochleb) [inline urdf/orbbec_astra.gazebo.xacro:3]: Parameters should match those from ROS 2

File: urdf/ouster_os1_32.gazebo.xacro [hunk 5]
```diff
@@ -0,0 +1,93 @@
+<?xml version="1.0"?>
+<robot xmlns:xacro="http://www.ros.org/wiki/xacro">
+  <xacro:macro name="ouster_os1_32" params="xyz rpy use_gpu:=false prefix:=front topic:=laser/scan">
```
Comment (Kotochleb) [inline urdf/ouster_os1_32.gazebo.xacro:3]: Parameters should match those from ROS 2

File: urdf/rplidar_s1.gazebo.xacro [hunk 6]
```diff
@@ -0,0 +1,86 @@
+<?xml version="1.0"?>
+<robot xmlns:xacro="http://www.ros.org/wiki/xacro">
+  <xacro:macro name="rplidar_s1" params="xyz rpy use_gpu:=false prefix:=front topic:=laser/scan">
```
Comment (Kotochleb) [inline urdf/rplidar_s1.gazebo.xacro:3]: Parameters should match those from ROS 2

File: urdf/velodyne_puck.gazebo.xacro [hunk 7]
```diff
@@ -0,0 +1,85 @@
+<?xml version="1.0"?>
+<robot xmlns:xacro="http://www.ros.org/wiki/xacro">
+  <xacro:macro name="velodyne_puck" params="xyz rpy use_gpu:=false prefix:=front topic:=laser/scan">
```
Comment (Kotochleb) [inline urdf/velodyne_puck.gazebo.xacro:3]: Parameters should match those from ROS 2

**Review by Kotochleb — CHANGES_REQUESTED** [review]

File: urdf/velodyne_puck.gazebo.xacro [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+
+    <gazebo reference="${tf_prefix_ext}${frame_id}">
+      <sensor type="ray" name="${tf_prefix_ext}velodyne_puck_sensor">
+        <pose>0 0 0.0378 0 0 0</pose>
+        <visualize>false</visualize>
+        <update_rate>10</update_rate>
+        <ray>
+          <scan>
+            <horizontal>
+              <samples>3600</samples>
```
Comment (pkowalsk1) [inline urdf/velodyne_puck.gazebo.xacro:73]: In my case, this is the maximum value. More samples cause an error: 'Spawn service failed. Exiting.'

File: urdf/orbbec_astra.gazebo.xacro [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+      <xacro:property name="tf_prefix_ext" value="${tf_prefix}_" />
+    </xacro:unless>
+
+    <joint name="${parent_link}_to_${tf_prefix_ext}orbbec_astra_link_joint" type="fixed">
+      <origin xyz="${xyz}" rpy="${rpy}" />
+      <parent link="${parent_link}" />
+      <child link="${tf_prefix_ext}orbbec_astra_link" />
+    </joint>
+
+    <link name="${tf_prefix_ext}orbbec_astra_link">
```
Comment (pkowalsk1) [inline urdf/orbbec_astra.gazebo.xacro:24]: This frames orientation change was needed becouse depth image was rotated.

File: package.xml [hunk 10]
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0"?>
+<package format="3">
+  <name>ros_components_description</name>
+  <version>0.1.0</version>
+  <description>URDF descriptions of components offered with Husarion robots</description>
+
+  <maintainer email="support@husarion.com">Husarion</maintainer>
+  <maintainer email="krzysztof.wojciechowski@husarion.com">Krzysztof Wojciechowski</maintainer>
+  <maintainer email="pawel.kowalski@husarion.com">Pawel Kowalski</maintainer>
```
Comment (KmakD) [inline package.xml:9]:
```suggestion
  <author email="krzysztof.wojciechowski@husarion.com">Krzysztof Wojciechowski</author>
  <author email="pawel.kowalski@husarion.com">Pawel Kowalski</author>
  <maintainer email="support@husarion.com">Husarion</maintainer>
```
I would mark people as authors

File: package.xml [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+
+  <url type="website">https://husarion.com/</url>
+  <url type="repository">https://github.com/husarion/panther_description/tree/ros1</url>
+  <url type="bugtracker">https://github.com/husarion/panther_description/issues</url>
+
+  <buildtool_depend>catkin</buildtool_depend>
+
+  <exec_depend>gazebo_ros</exec_depend>
+  <exec_depend>urdf</exec_depend>
+  <exec_depend>xacro</exec_depend>
```
Comment (KmakD) [inline package.xml:20]:
```suggestion
  <depend>gazebo_ros</depend>
  <depend>urdf</depend>
  <depend>xacro</depend>
```

File: urdf/orbbec_astra.gazebo.xacro [hunk 12]
```diff
@@ -0,0 +1,102 @@
...
+      <collision>
+        <geometry>
+          <box size="0.165 0.04 0.05" />
+        </geometry>
+        <origin xyz="-0.01 0 0.025" rpy="0 0 ${pi/2}"/>
+      </collision>
+
+      <inertial>
+          <mass value="0.105" />
+          <inertia ixx="1" ixy="0" ixz="0" iyy="1" iyz="0" izz="1" />
```
Comment (KmakD) [inline urdf/orbbec_astra.gazebo.xacro:41]:
```suggestion
          <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0" />
```

File: urdf/orbbec_astra.gazebo.xacro [hunk 13]
```diff
@@ -0,0 +1,102 @@
...
+      </inertial>
+    </link>
+
+    <joint name="${tf_prefix_ext}${frame_id}_link_joint" type="fixed">
+      <origin xyz="0.01 0 0.035" rpy="${-pi/2} 0 ${-pi/2}" />
+      <parent link="${tf_prefix_ext}orbbec_astra_link" />
+      <child link="${tf_prefix_ext}${frame_id}" />
+    </joint>
+
+    <link name="${tf_prefix_ext}${frame_id}"></link>
```
Comment (KmakD) [inline urdf/orbbec_astra.gazebo.xacro:51]:
```suggestion
    <link name="${tf_prefix_ext}${frame_id}" />
```

File: urdf/ouster_os1_32.gazebo.xacro [hunk 14]
```diff
@@ -0,0 +1,99 @@
...
+    <joint name="${parent_link}_to_${tf_prefix_ext}ouster_os1_32_link_joint" type="fixed">
+      <origin xyz="${xyz}" rpy="${rpy}" />
+      <parent link="${parent_link}" />
+      <child link="${tf_prefix_ext}ouster_os1_32_link" />
+    </joint>
+
+    <link name="${tf_prefix_ext}ouster_os1_32_link">
+      <visual>
+        <geometry>
+          <mesh filename="file://$(find ros_components_description)/meshes/ouster_os1_32.stl" scale="1 1 0.65" />
```
Comment (KmakD) [inline urdf/ouster_os1_32.gazebo.xacro:26]:
```suggestion
          <mesh filename="file://$(find ros_components_description)/meshes/ouster_os1_32.stl" scale="1.0 1.0 0.65" />
```
Also, scale in z seems weird to me, is it intentional?

File: urdf/ouster_os1_32.gazebo.xacro [hunk 15]
```diff
@@ -0,0 +1,99 @@
...
+      <collision>
+        <geometry>
+          <cylinder radius="${0.0753/2.0}" length="0.0435" />
+        </geometry>
+        <origin xyz="0.0 0.0 ${0.0435/2.0}" rpy="0.0 0.0 0.0"/>
+      </collision>
+
+      <inertial>
+          <mass value="0.447" />
+          <inertia ixx="1" ixy="0" ixz="0" iyy="1" iyz="0" izz="1" />
```
Comment (KmakD) [inline urdf/ouster_os1_32.gazebo.xacro:41]:
```suggestion
          <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0" />
```

File: urdf/ouster_os1_32.gazebo.xacro [hunk 16]
```diff
@@ -0,0 +1,99 @@
...
+      </inertial>
+    </link>
+
+    <joint name="${tf_prefix_ext}ouster_os1_32_link_to_${tf_prefix_ext}${frame_id}_joint" type="fixed">
+      <origin xyz="0 0 0.023" rpy="0 0 0" />
```
Comment (KmakD) [inline urdf/ouster_os1_32.gazebo.xacro:46]:
```suggestion
      <origin xyz="0.0 0.0 0.023" rpy="0.0 0.0 0.0" />
```

File: urdf/ouster_os1_32.gazebo.xacro [hunk 17]
```diff
@@ -0,0 +1,99 @@
...
+      <parent link="${tf_prefix_ext}ouster_os1_32_link" />
+      <child link="${tf_prefix_ext}${frame_id}" />
+    </joint>
+
+    <link name="${tf_prefix_ext}${frame_id}"></link>
```
Comment (KmakD) [inline urdf/ouster_os1_32.gazebo.xacro:51]:
```suggestion
    <link name="${tf_prefix_ext}${frame_id}" />
```

File: urdf/rplidar_s1.gazebo.xacro [hunk 18]
```diff
@@ -0,0 +1,100 @@
...
+      <parent link="${parent_link}" />
+      <child link="${tf_prefix_ext}rplidar_s1_link" />
+    </joint>
+
+    <link name="${tf_prefix_ext}rplidar_s1_link">
+      <visual>
+        <geometry>
+          <mesh filename="file://$(find ros_components_description)/meshes/rplidar_s1.stl" scale="0.0001 0.0001 0.0001" />
+        </geometry>
+        <origin xyz="0.0 0.0 ${0.051/2.0}" rpy="0 0 -${pi/2.0}"/>
```
Comment (KmakD) [inline urdf/rplidar_s1.gazebo.xacro:38]:
```suggestion
        <origin xyz="0.0 0.0 ${0.051/2.0}" rpy="0.0 0.0 -${pi/2.0}"/>
```

File: urdf/rplidar_s1.gazebo.xacro [hunk 19]
```diff
@@ -0,0 +1,100 @@
...
+      <collision>
+        <geometry>
+          <cylinder radius="${0.0555/2.0}" length="0.051" />
+        </geometry>
+        <origin xyz="0.0 0.0 ${0.051/2.0}" rpy="0.0 0.0 0.0"/>
+      </collision>
+
+      <inertial>
+        <mass value="0.105" />
+        <inertia ixx="1" ixy="0" ixz="0" iyy="1" iyz="0" izz="1" />
```
Comment (KmakD) [inline urdf/rplidar_s1.gazebo.xacro:51]:
```suggestion
        <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0" />
```

File: urdf/rplidar_s1.gazebo.xacro [hunk 20]
```diff
@@ -0,0 +1,100 @@
...
+      </inertial>
+    </link>
+
+    <joint name="${tf_prefix_ext}rplidar_s1_to_${tf_prefix_ext}${frame_id}_joint" type="fixed">
+      <origin xyz="0 0 0.04" rpy="0.0 0.0 0.0" />
+      <parent link="${tf_prefix_ext}rplidar_s1_link" />
+      <child link="${tf_prefix_ext}${frame_id}" />
+    </joint>
+
+    <link name="${tf_prefix_ext}${frame_id}"></link>
```
Comment (KmakD) [inline urdf/rplidar_s1.gazebo.xacro:61]:
```suggestion
    <link name="${tf_prefix_ext}${frame_id}" />
```

File: urdf/velodyne_puck.gazebo.xacro [hunk 21]
```diff
@@ -0,0 +1,97 @@
...
+      <collision>
+        <geometry>
+          <cylinder radius="${0.090/2.0}" length="0.0717" />
+        </geometry>
+        <origin xyz="0.0 0.0 ${0.0717/2.0}" rpy="0.0 0.0 0.0"/>
+      </collision>
+
+      <inertial>
+        <mass value="0.830" />
+        <inertia ixx="1" ixy="0" ixz="0" iyy="1" iyz="0" izz="1" />
```
Comment (KmakD) [inline urdf/velodyne_puck.gazebo.xacro:41]:
```suggestion
        <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0" />
```

**Review by KmakD — CHANGES_REQUESTED** [review]

File: urdf/ouster_os1_32.gazebo.xacro [hunk 22]
```diff
@@ -0,0 +1,99 @@
...
+    <joint name="${parent_link}_to_${tf_prefix_ext}ouster_os1_32_link_joint" type="fixed">
+      <origin xyz="${xyz}" rpy="${rpy}" />
+      <parent link="${parent_link}" />
+      <child link="${tf_prefix_ext}ouster_os1_32_link" />
+    </joint>
+
+    <link name="${tf_prefix_ext}ouster_os1_32_link">
+      <visual>
+        <geometry>
+          <mesh filename="file://$(find ros_components_description)/meshes/ouster_os1_32.stl" scale="1.0 1.0 0.65" />
```
Comment (KmakD) [inline urdf/ouster_os1_32.gazebo.xacro:26]:
```suggestion
          <mesh filename="file://$(find ros_components_description)/meshes/ouster_os1_32.stl" scale="1.0 1.0 1.0" />
```
scale in z seems a bit weird

**Review by KmakD — CHANGES_REQUESTED** [review]

**Review by KmakD — APPROVED** [review]

**Pull request merged**